### PR TITLE
MINOR: [C++][Azure][FS] Document some limitations and atomicity guarantees

### PR DESCRIPTION
**Pull request by felipecrv**
### Rationale for this change

Documenting some details of the behavior of destructive filesystem operations.

### What changes are included in this PR?

Only docstring changes.

### Are these changes tested?

N/A.

### CODE REVIEW

**Comment by felipecrv** [comment]
@Tom-Newton

File: cpp/src/arrow/filesystem/azurefs.h [hunk 1]
```diff
@@ -264,15 +264,35 @@ class ARROW_EXPORT AzureFileSystem : public FileSystem {
...
+  /// implemented for safety reasons).
+  ///
+  /// \return Status::NotImplemented
   Status DeleteRootDirContents() override;
 
+  /// \brief Deletes a file.
+  ///
+  /// Supported on both flat namespace and Hierarchical Namespace storage
+  /// accounts. A check is made to guarantee the parent directory doesn't
+  /// disappear after the blob is deleted and while this operation is running,
```
Comment (Tom-Newton) [inline cpp/src/arrow/filesystem/azurefs.h:288]: Is this referring to flat namespace accounts only? I can't imagine a hierarchical namespace directory disappearing when a file is deleted.

**Review by Tom-Newton — APPROVED** [review]
One comment but it looks go to me

File: cpp/src/arrow/filesystem/azurefs.h (hunk 1 above)
Comment (felipecrv) [reply cpp/src/arrow/filesystem/azurefs.h:288]: Re-wording suggestion? My intention is being concise, but yeah, none of this applies to HNS accounts.

Comment (Tom-Newton) [reply cpp/src/arrow/filesystem/azurefs.h:288]: Less concise but I would go for 
```suggestion
  /// accounts. On flat namespace storage accounts a check is made to guarantee the parent directory doesn't
  /// disappear after the blob is deleted and while this operation is running,
```
I don't feel strongly

**Review by kou — APPROVED** [review]
+1

**Pull request merged**